Add support for requestPermissions option

simonbelanger opened the pull request:
Required to specify scope for multi-tenants app

simonbelanger committed:
Add support for requestPermissions option

Required to specify scope for multi-tenants app

djluck commented:
Looks good. I'll try to merge and push a new version in the next couple of days.

simonbelanger committed:
Replaced Azure AD Graph by Microsoft Graph as the base resource for the azureAd service

simonbelanger commented:
In our case, it will be more useful to store only an access token for the unified API exposed by the Microsoft Graph resource.

The `https://graph.microsoft.com/v1.0/me` endpoint is the equivalent of `https://graph.windows.net/myorganization/me?api-version=2013-11-08` (return nearly the same data and doesn't require any special scope or permission from the user).

I know it's an important change, np if you want to skip it.

simonbelanger committed:
Fixed typo in AzureAd.whitelistedFields

djluck commented:
Hey Simon, looks like someone has been busy! It's been a long time I've looked at this project (or done any meteor work at all) so reviewing and releasing may take a while. Thanks for fixing 
#5 as well!

djluck commented:
Simon,

I don't have any idea what the new graph.microsoft.com endpoint offers over the graph.windows.net endpoint, are there any benefits to using it?
This might introduce a breaking change for users who authenticated against graph.windows.net- the access and refresh tokens stored in the database will probably not work against graph.microsoft.com. Have you been able to test that?

Cheers,

James

simonbelanger commented:
From Azure AD Graph API (graph.windows.net) documentation : 

> Azure AD Graph API functionality is also available through Microsoft Graph, a unified API that also includes APIs from other Microsoft services like Outlook, OneDrive, OneNote, Planner, and Office Graph, all accessed through a single endpoint with a single access token.

Azure AD Graph API and Microsoft Graph are two different resources and they both need their own access token stored on the user. So with this change, a user would no longer be able to access graph.windows.net by using the access token stored in services.azureAd.accessToken.

But the same refresh token can be used to get an access token for both resource.

Zertz committed:
Add support for redirectUrl

This allows using loginStyle: "redirect" and redirectUrl as a way to save state between the redirects.

simonbelanger committed:
Merge pull request #1 from Zertz/patch-3

Add support for redirectUrl
